Session/Shared/OWSQRCodeScanningViewController.m

```diff
@@ -115,7 +115,10 @@ - (void)startCapture
                 self.capture.layer.frame = self.view.bounds;
                 [self.view.layer addSublayer:self.capture.layer];
                 [self.view bringSubviewToFront:self.maskingView];
-                [self.capture start];
+                
+                dispatch_async(self.captureQueue, ^{
+                    [self.capture start];
+                });
```

Comment:
Does this need to be in the view hierarchy before we start capturing? (just wondering if we can move the `[self.capture start];` to be above the `dispatch_async(dispatch_get_main_queue()` since it's already on the right queue at that point)

Reply:
Not sure about this, I'll have a try.

Reply:
We can move the code. And I also found `ZXingObjC` seems to be calling `[AVCaptureSession startRunning]` on main thread, which can block the UI...

Reply:
That's annoying...